hypersync-format/src/types/quantity.rs

```diff
@@ -106,14 +106,36 @@ impl Visitor<'_> for QuantityVisitor {
 
         Ok(Quantity::from(buf))
     }
+
+    fn visit_u64<E>(self, value: u64) -> StdResult<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        // Convert the integer to big-endian bytes and canonicalize
+        let buf = canonicalize_bytes(value.to_be_bytes().to_vec());
+        Ok(Quantity::from(buf))
+    }
+
+    fn visit_i64<E>(self, value: i64) -> StdResult<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        if value < 0 {
+            return Err(E::custom("negative quantity not allowed"));
+        }
+        let unsigned = value as u64;
+        let buf = canonicalize_bytes(unsigned.to_be_bytes().to_vec());
+        Ok(Quantity::from(buf))
+    }
 }
 
 impl<'de> Deserialize<'de> for Quantity {
     fn deserialize<D>(deserializer: D) -> StdResult<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        deserializer.deserialize_str(QuantityVisitor)
+        // Accept either hex strings (0x...) or numeric JSON values
+        deserializer.deserialize_any(QuantityVisitor)
```

Comment:
Not a big deal but probably would be better to call deserialize_str first, match and if there's an error call `deserialize_u64`

Then there's no overhead in it just running all the visit functions. You control the entry point. And you don't need visit_i64 for eg.

Reply:
I actually think this would be a worthwhile thing to do since we use quantity for a lot of fields